src/dash/WebmSegmentBaseLoader.js

```diff
@@ -0,0 +1,413 @@
+
+                // block number is strictly optional.
+                // we also have to make sure we don't go beyond the end
+                // of the cues
+                if (ebmlParser.getPos() + 4 > cueSize ||
+                        !ebmlParser.consumeTag(WebM.Segment.Cues.CuePoint.CueTrackPositions.CueBlockNumber, true)) {
+                    cue.CueTracks.push(cueTrack);
+                } else {
+                    // since we have already consumed the tag, get the size of
+                    // the tag's payload, and manually parse an unsigned int
+                    // from the bit stream
+                    numSize = ebmlParser.getMatroskaCodedNum();
+                    cueTrack.BlockNumber = ebmlParser.getMatroskaUint(numSize);
+
+                    cue.CueTracks.push(cueTrack);
+                }
+            }
+
+            if (cue.CueTracks.length === 0) {
+                throw new Error('Mandatory cuetrack not found');
+            }
+            cues.push(cue);
+        }
+
+        if (cues.length === 0) {
+            throw new Error('mandatory cuepoint not found');
+        }
+        return cues;
+    }
+
+    function parseSegments(data, media, segmentStart, segmentEnd, segmentDuration) {
+        let duration;
+        let parsed;
+        let segments;
+        let segment;
+        let i;
+        let len;
+        let start;
+        let end;
+
+        parsed = parseCues(data);
+        segments = [];
+
+        // we are assuming one cue track per cue point
+        // both duration and media range require the i + 1 segment
+        // the final segment has to use global segment parameters
+        for (i = 0, len = parsed.length; i < len; i += 1) {
+            segment = new Segment();
+            duration = 0;
+
+            if (i < parsed.length - 1) {
+                duration = parsed[i + 1].CueTime - parsed[i].CueTime;
+            } else {
+                duration = segmentDuration - parsed[i].CueTime;
+            }
+
+            segment.duration = duration;
+            segment.media = media;
+            segment.startTime = parsed[i].CueTime;
+            segment.timescale = 1000; // hardcoded for ms
+            start = parsed[i].CueTracks[0].ClusterPosition + segmentStart;
+
+            if (i < parsed.length - 1) {
+                end = parsed[i + 1].CueTracks[0].ClusterPosition + segmentStart - 1;
+            } else {
+                end = segmentEnd - 1;
+            }
+
+            segment.mediaRange = start + '-' + end;
+            segments.push(segment);
+        }
+
+        log('Parsed cues: ' + segments.length + ' cues.');
+
+        return segments;
+    }
+
+    function parseEbmlHeader(data, media, theRange, callback) {
+        let ebmlParser = EBMLParser(context).create({data: data});
+        let duration;
+        let segments;
+        let parts = theRange.split('-');
+        let request = new XMLHttpRequest();
```

Comment:
@bbcrddave good point. I'll make it use the XHRLoader, as segment base loaders are already some kind of derived loaders, and update the PR.